eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/source/connector/HttpSourceConnector.java

```diff
@@ -83,15 +82,27 @@ private void doInit() {
             .handler(LoggerHandler.create())
             .handler(ctx -> {
                 VertxMessageFactory.createReader(ctx.request())
-                    .map(MessageReader::toEvent)
+                    .map(reader -> {
+                        CloudEvent event = reader.toEvent();
+                        if (event.getSubject() == null) {
+                            throw new IllegalStateException("attribute 'subject' cannot be null");
+                        }
+                        if (event.getDataContentType() == null) {
+                            throw new IllegalStateException("attribute 'datacontenttype' cannot be null");
+                        }
+                        if (event.getData() == null) {
+                            throw new IllegalStateException("attribute 'data' cannot be null");
+                        }
+                        return event;
+                    })
                     .onSuccess(event -> {
                         queue.add(event);
                         log.info("[HttpSourceConnector] Succeed to convert payload into CloudEvent. StatusCode={}", HttpResponseStatus.OK.code());
                         ctx.response().setStatusCode(HttpResponseStatus.OK.code()).end();
                     })
                     .onFailure(t -> {
                         log.error("[HttpSourceConnector] Malformed request. StatusCode={}", HttpResponseStatus.BAD_REQUEST.code(), t);
-                        ctx.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code()).setStatusMessage(t.getMessage()).end();
+                        ctx.response().setStatusCode(HttpResponseStatus.BAD_REQUEST.code()).end();
                     });
```

Comment:
Why default err message is used instead of "attribute 'subject' cannot be null"?

Reply:
> Why default err message is used instead of "attribute 'subject' cannot be null"?

I think it's not so good to place error message here. In some cases, such as lack of attribute `source`, we get an error message like this from `reader.toEvent()`.
```
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Missing mandatory source attribute
 at [Source: (byte[])"{
    "specversion": "1.0",
    "type":"com.example.someevent",
    "subject":"test_topic",
    "datacontenttype":"text/plain",
    "id": "A234-1234-1234",
    "data": "aa"
}"; line: 8, column: 1]
```
 It contains '\r\n' which cannot be used in the http status message and cause another exception in `setStatusMessage`.
```
2023-12-16 16:18:05,378 ERROR [vert.x-eventloop-thread-0] ContextBase(:) - Unhandled exception
java.lang.IllegalArgumentException: reasonPhrase contains one of the following prohibited characters: \r\n: com.fasterxml.jackson.databind.exc.MismatchedInputException: Missing mandatory source attribute
 at [Source: (byte[])"{
    "specversion": "1.0",
    "type":"com.example.someevent",
    "subject":"test_topic",
    "datacontenttype":"text/plain",
    "id": "A234-1234-1234",
    "data": "aa"
}"; line: 8, column: 1]
	at io.netty.handler.codec.http.HttpResponseStatus.<init>(HttpResponseStatus.java:558) ~[netty-codec-http-4.1.100.Final.jar:4.1.100.Final]
	at io.netty.handler.codec.http.HttpResponseStatus.<init>(HttpResponseStatus.java:546) ~[netty-codec-http-4.1.100.Final.jar:4.1.100.Final]
	at io.vertx.core.http.impl.Http1xServerResponse.setStatusMessage(Http1xServerResponse.java:166) ~[vertx-core-4.4.6.jar:4.4.6]
	at org.apache.eventmesh.connector.http.source.connector.HttpSourceConnector.lambda$null$2(HttpSourceConnector.java:105) ~[main/:?]
```

As a result, the server cannot response to the sender.